python/ray/actor.py

```diff
@@ -204,8 +204,55 @@ class ActorClass(object):
             each actor method.
     """
 
-    def __init__(self, modified_class, class_id, max_reconstructions, num_cpus,
-                 num_gpus, memory, object_store_memory, resources):
+    def __init__(cls, name, bases, attr):
+        """Prevents users from directly inheriting from an ActorClass.
+
+        This will be called when a class is defined with an ActorClass object
+        as one of its base classes. To intentionally construct an ActorClass,
+        use the 'from_modified_class' classmethod.
+
+        Raises:
+            TypeError: Always.
+        """
+        for base in bases:
+            if isinstance(base, ActorClass):
+                raise TypeError("Attempted to define subclass '{}' of actor "
+                                "class '{}'. Inheriting from actor classes is "
+                                "not currently supported. You can instead "
+                                "inherit from a non-actor base class and make "
+                                "the derived class an actor class (with "
+                                "@ray.remote).".format(name, base._class_name))
+
+        # This shouldn't be reached because one of the base classes must be
+        # an actor class if this was called, but raise a more generic error
+        # just in case.
+        raise TypeError("Attempted to define subclass '{}' of an actor class. "
+                        "Inheriting from actor classes is not currently "
+                        "supported. You can instead inherit from a non-actor "
+                        "base class and make the derived class an actor class "
+                        "(with @ray.remote).".format(name, base._class_name))
```

Comment:
`base` doesn't exist here, right?

Is it possible to reach this code if the user does something really weird? Just wondering if we should raise the error you have or just have an `assert False`.

Reply:
Ah good catch. I'm honestly not sure if it's possible to reach this code via any reasonable behavior. On further reflection it's probably better to `assert False` because this error message could be misleading.